File: app/src/main/java/com/zac4j/system/ui/MainActivity.java
package com.zac4j.system.ui;

import android.content.DialogInterface;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Button;
import android.widget.ImageView;
import com.zac4j.system.R;
import com.zac4j.system.util.screenshot.ScreenshotHelper;

public class MainActivity extends AppCompatActivity {

    private ImageView mImageView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        mImageView = findViewById(R.id.imageView);

        FloatingActionButton fab = findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                showDialog();

                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                    .setAction("Action", null).show();
            }
        });
    }

    private void showDialog() {
        final AlertDialog alertDialog = new AlertDialog.Builder(this)
            .setMessage("Trump nmsl!")
            .setPositiveButton("OK", null)
            .setNegativeButton("Cancel", null)
            .create();

        alertDialog.setOnShowListener(new DialogInterface.OnShowListener() {
            @Override public void onShow(DialogInterface dialog) {
                Button okBtn = alertDialog.getButton(AlertDialog.BUTTON_POSITIVE);
                okBtn.setOnClickListener(new View.OnClickListener() {
                    @Override public void onClick(View v) {
                        Bitmap bitmap = ScreenshotHelper.takeScreenshot(MainActivity.this);
                        mImageView.setImageBitmap(bitmap);
                    }
                });
            }
        });

        alertDialog.show();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/zac4j/system/ui/MainActivity.kt
package com.zac4j.system.ui

import android.os.Bundle
import android.view.MotionEvent
import android.view.View
import android.widget.ImageView
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AlertDialog.Builder
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider.AndroidViewModelFactory
import com.google.android.material.snackbar.Snackbar
import com.zac4j.system.R
import com.zac4j.system.R.layout
import com.zac4j.system.data.MainViewModel
import com.zac4j.system.databinding.ActivityMainBinding
import com.zac4j.system.util.screenshot.ScreenshotHelper

class MainActivity : AppCompatActivity() {

  companion object {
    private const val TAG = "MainActivity"
  }

  private val viewModel by lazy {
    AndroidViewModelFactory.getInstance(application)
        .create(MainViewModel::class.java)
  }

  private val mImageView: ImageView? = null
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    val binding: ActivityMainBinding = DataBindingUtil.setContentView(this, layout.activity_main)

    binding.viewModel = viewModel

    binding.lifecycleOwner = this

    setSupportActionBar(binding.toolbar)

    val networkPage = NetworkFragment()
    val touchPage = TouchFragment()
    val dialPage = DialFragment()
    supportFragmentManager.beginTransaction()
        .add(R.id.fragment_container, dialPage)
        .addToBackStack(null)
        .commit()

    binding.fab.setOnClickListener { view ->
      showSnackBar(view)
    }
  }

  private fun showSnackBar(view: View) {
    Snackbar.make(view, "Hi", Snackbar.LENGTH_LONG)
        .setAction("Action", null)
        .show()
  }

  private fun showDialog() {
    val alertDialog =
      Builder(this)
          .setMessage("!")
          .setPositiveButton("OK", null)
          .setNegativeButton("Cancel", null)
          .create()
    alertDialog.setOnShowListener {
      val okBtn =
        alertDialog.getButton(AlertDialog.BUTTON_POSITIVE)
      okBtn.setOnClickListener {
        val bitmap = ScreenshotHelper.takeScreenshot(this@MainActivity)
        mImageView!!.setImageBitmap(bitmap)
      }
    }
    alertDialog.show()
  }

  override fun onTouchEvent(ev: MotionEvent): Boolean {
//    Logger.i(
//        TAG, "onTouchEvent",
//        "MotionEvent::${MotionEvent.actionToString(ev.action)}"
//    )
    return super.onTouchEvent(ev)
  }

  override fun dispatchTouchEvent(ev: MotionEvent): Boolean {
//    Logger.i(
//        TAG, "dispatchTouchEvent",
//        "MotionEvent::${MotionEvent.actionToString(ev.action)}"
//    )
    return super.dispatchTouchEvent(ev)
  }
}